documentation/src/docs/wire-protocol.md

```diff
@@ -11,102 +11,97 @@ You may obtain a copy of the License at
 
-All requests and replies have an 8 byte header with two fields, (All data is written in BigEndian format):
-1.  Message Type - An integer (4 bytes) identifies the message type and this determines what fields will follow. (Note the protocol can be extended by adding new types)
-2.  Length - Unsigned integer 4 bytes (Messages should be &lt; 2^24, but the upper bits remain zero). How many bytes of data from this point forward are part of this message. (Possibly zero, indicating there is no data)
+All the requests and replies have an 8 bytes header with two fields.(All data is written in **BigEndian** format).
 
+- **Message Type**: An Integer of 4 bytes identifies the message type and determines what fields will follow. (Note that the protocol can be extended by adding new types.)
+- **Length**:  Unsigned Integer of 4 bytes (Messages should be less than 2^<sup>24</sup>, but the upper bits remain zero). The remaining bits followed by the upper bit are part of the message. (Possibly zero, indicating there is no data).
 The remainder of the fields are specific to the type of message. A few important messages are listed below.
 
-General
--------
+# General
 
-### Partial Message - Request/Reply
+## Partial Message - Request/Reply
 
-1.  Begin/Middle/End - Enum (1 byte)
-2.  Data
-A partial message is one that was broken up when being sent over the wire. (For any reason). The whole message is reconstructed by reading the partial messages in sequence and assembling them into a whole. It is not valid to attempt to start a new partial message before completing the previous one.
+-  **Begin/Middle/End**: Enum (1 byte)
+-  **Data:** A partial message is the one that was broken up when the message was sent over the wire. It is essential that at any cause, the whole message is reconstructed by reading the partial messages in sequence and assembling them into a whole. It is not valid to attempt to start a new partial message before completing the previous one.
 
-### KeepAlive - Request/Reply
+## KeepAlive - Request/Reply
 
-1.  Data - uninterpreted data of the length of the message. (Usually 0 bytes)
+- **Data**: Uninterpreted data of the length of the message. (Usually 0 byte)
 
-Reading
--------
+# Reading
 
-### Read Segment - Request
 
-1.  Segment to read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset to read from - Long (8 bytes)
-3.  Suggested Length of Reply - int (4 bytes)
-    1.  This is how much data the client wants. They won't necessarily get that much.
+## Read Segment - Request
 
-### Segment Read - Reply
+1. Segment to read: String (2 bytes length, followed by that many bytes of Java's Modified UTF-8).
+2. Offset to read from:  Long (8 bytes).
+3. Suggested Length of Reply: Integer (4 bytes). This is based on the clients requirements. (They won't necessarily get that much).
```

Comment:
@tkaitchuck `This is based on the clients requirements. (They won't necessarily get that much).` 
need clarity???

Reply:
There is a request length, this is how many bytes the client wants. The server may ignore this and give it a different number of bytes. It is treated as a suggestion to the server.

Reply:
Accepted suggestion